frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/SlideOut.java

```diff
@@ -31,125 +32,100 @@
 import com.virtuslab.gitmachete.backend.api.IManagedBranchSnapshot;
 import com.virtuslab.gitmachete.frontend.actions.dialogs.DeleteBranchOnSlideOutSuggestionDialog;
 import com.virtuslab.gitmachete.frontend.file.MacheteFileWriter;
-import com.virtuslab.gitmachete.frontend.ui.api.table.BaseEnhancedGraphTable;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
 import com.virtuslab.qual.async.ContinuesInBackground;
 import com.virtuslab.qual.guieffect.UIThreadUnsafe;
 
 @CustomLog
-// For some reason, using `@ExtensionMethod({GitMacheteBundle.class})` on this class
-// leads to weird Checker Framework errors :/
-public class SlideOutBackgroundable extends Task.Backgroundable {
+@ExtensionMethod({GitMacheteBundle.class})
+public class SlideOut {
 
   private final Project project;
   private final String branchToSlideOutName;
   @Nullable
   private final String currentBranchNameIfManaged;
   private final BranchLayout branchLayout;
   private final GitRepository gitRepository;
-  private final BaseEnhancedGraphTable graphTable;
-  private final @UI Runnable doInUIThreadWhenReady;
 
   public static final String DELETE_LOCAL_BRANCH_ON_SLIDE_OUT_GIT_CONFIG_KEY = "machete.slideOut.deleteLocalBranch";
 
-  public SlideOutBackgroundable(String title, IManagedBranchSnapshot branchToSlideOut,
+  public SlideOut(IManagedBranchSnapshot branchToSlideOut,
```

Comment:
The problem with the naming now... is that this class is still _kinda a backgroundable_, as part of the workload (deleting the branch, getting/setting config) happens in backgroundables 😅 but this should be apparent from `@ContinuesInBackground` on `run`